### PR TITLE
Fixed export URL

### PR DESCRIPTION
**Pull request by assertchris**

### CODE REVIEW

**Comment by tractorcow** [comment]
URLs aren't visible in the report view anymore.

The exported URLs should probably be absolute urls, since a CSV doesn't have a base_url like the CMS does.

**Comment by assertchris** [comment]
> URLs aren't visible in the report cms view anymore.

This was intentional, but I can see a reason for wanting it. Will add back.

> The CSV exported URLs should probably be absolute urls, since a CSV doesn't have a base_url like the CMS does.

Sure - I'll fix.

File: code/SitewideContentReport.php [hunk 1]
```diff
@@ -235,6 +243,17 @@ public function getDataFieldValue($record, $fieldName)
             return $callback($record);
         }
 
+        // Links
+        if ($fieldName == "Link") {
+
+            return Controller::join_links(Director::absoluteBaseURL(), $record->Link());
```
Comment (tractorcow) [inline code/SitewideContentReport.php]: The problem is that there's a risk that baseurl is joined twice:

`Director::absoluteBaseURL` is the protocol + hostname + basepath
`$record->Link()` is basepath + relative url.


image


Instead shall we use `$record->AbsoluteLink()` which is on both File and SiteTree?

File: code/SitewideContentReport.php [hunk 2]
```diff
@@ -235,6 +243,17 @@ public function getDataFieldValue($record, $fieldName)
...
+        }
+
+        // URLSegments
+        if ($fieldName == "URL" || $fieldName == "URLSegment") {
+            return str_replace(BASE_URL, "", $record->Link());
```
Comment (tractorcow) [inline code/SitewideContentReport.php]: You can use `$record->RelativeLink()` and save having to refer to BASE_URL. :D

**Pull request merged**